aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/HttpConnectorConfigCustomization.kt

```diff
@@ -150,18 +154,33 @@ class HttpConnectorConfigCustomization(
                     /// ## }
                     /// ## }
                     /// ```
-                    pub fn set_http_connector(&mut self, http_connector: Option<impl Into<#{HttpConnector}>>) -> &mut Self {
-                        self.http_connector = http_connector.map(|inner| inner.into());
-                        self
-                    }
                     """,
                     *codegenScope,
                 )
-            }
-            is ServiceConfig.BuilderBuild -> writable {
                 if (runtimeMode.defaultToOrchestrator) {
-                    rust("layer.store_or_unset(self.http_connector);")
+                    rustTemplate(
+                        """
+                        pub fn set_http_connector(&mut self, http_connector: #{Option}<impl #{Into}<#{HttpConnector}>>) -> &mut Self {
+                            http_connector.map(|c| self.inner.store_put(c.into()));
```

Comment:
Should we be using `ConfigBagAccessors` to set these directly to ensure the types match on both ends?

Reply:
Agreed, that's what we should. We actually use `ConfigBagAccessors` already in [ServiceRuntimePluginGenerator](https://github.com/awslabs/smithy-rs/blob/main/codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/ServiceRuntimePluginGenerator.kt#LL165-L165) for setting up a connection, and that line of code will be moved to builder's build method in the next PR (by which time we are using `ConfigBagAccessors` in the `build` method to set up a `connection` when a service config is created). Within `set_http_connector`, I believe we can continue using `store_put` because "the other end" is right under its nose, e.g. the `build` method.